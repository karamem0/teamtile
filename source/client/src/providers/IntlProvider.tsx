//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import {
  IntlProvider as Provider,
  createIntl,
  MessageFormatElement
} from 'react-intl';

import en from '../translations/compiled/en.json';
import ja from '../translations/compiled/ja.json';

const translations: { [key: string]: Record<string, MessageFormatElement[]> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language.substring(0, 2)]
});

function IntlProvider(props: React.PropsWithChildren<unknown>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
