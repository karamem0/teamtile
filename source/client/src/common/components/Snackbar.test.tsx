//
// Copyright (c) 2021-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import IntlProvider from '../../providers/IntlProvider';
import { MessageBarIntent } from '@fluentui/react-components';
import Presenter from './Snackbar.presenter';
import { render } from '@testing-library/react';

it('should create a shapshot when the text parameter is undefined', () => {
  const params = {
    intent: undefined,
    text: undefined
  };
  const { asFragment } = render(
    <IntlProvider>
      <Presenter {...params} />
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should create a shapshot when the text parameter is error', () => {
  const params = {
    intent: 'error' as MessageBarIntent,
    text: 'error'
  };
  const { asFragment } = render(
    <IntlProvider>
      <Presenter {...params} />
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should create a shapshot when the text parameter is warning', () => {
  const params = {
    intent: 'warning' as MessageBarIntent,
    text: 'warning'
  };
  const { asFragment } = render(
    <IntlProvider>
      <Presenter {...params} />
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should create a shapshot when the text parameter is success', () => {
  const params = {
    intent: 'success' as MessageBarIntent,
    text: 'success'
  };
  const { asFragment } = render(
    <IntlProvider>
      <Presenter {...params} />
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
