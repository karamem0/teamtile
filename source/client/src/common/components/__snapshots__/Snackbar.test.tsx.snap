// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create shapshot of when text is undefined 1`] = `<DocumentFragment />`;

exports[`create shapshot of when type is error 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 1001;
  margin: 0.5rem;
}

.emotion-1 {
  width: 1rem;
  height: 1rem;
}

<div
    class="emotion-0"
  >
    <div
      class="fui-Alert ___co0mdz0_q48w6e0 f22iagw f122n59 f5pgtk9 f1g0x7ka f11qrl6u f1qch9an fjlbh76 fff7au0 f1bjk9e1 fwsfkhu f8wkphi f5ogflp f1hqa2wf f1f09k3d finvdd3 fzkkow9 fcdblym fg706s2 fjik90z fghlq4f f1gn591s fb073pr fjscplz fz58gqq fkhj508 fl43uef f19n0e5 fxugw4r"
      role="alert"
    >
      <span
        class="fui-Alert__icon ___1l1smhh_k5ukh20 fd461yt f4ybsrx f1g0x7ka f19lj068 f1qch9an f1cnd47f f1whyuy6"
      >
        <svg
          aria-hidden="true"
          class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
          fill="currentColor"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 2a8 8 0 1 1 0 16 8 8 0 0 1 0-16ZM7.8 7.11a.5.5 0 0 0-.63.06l-.06.07a.5.5 0 0 0 .06.64L9.3 10l-2.12 2.12-.06.07a.5.5 0 0 0 .06.64l.07.06c.2.13.47.11.64-.06L10 10.7l2.12 2.12.07.06c.2.13.46.11.64-.06l.06-.07a.5.5 0 0 0-.06-.64L10.7 10l2.12-2.12.06-.07a.5.5 0 0 0-.06-.64l-.07-.06a.5.5 0 0 0-.64.06L10 9.3 7.88 7.17l-.07-.06Z"
            fill="currentColor"
          />
        </svg>
      </span>
      error
      <button
        class="fui-Button rlr4yyk fui-Alert__action ___jw5j7g0_17p02te f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu f16muhyy fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 fwa4z56 fe5wyld ffrv2ww f1rnfj6r f9dg1pr f4xjyn1 f19ot1t3 fjaevkr f1yrt24w f1lym4af f11cxmw1 f9ddjv3 f1sbtcvk f81rol6 fdghr9 frdkuqy fy77jfu f44c6la fcgxt0o"
        title="Dismiss"
        type="button"
      >
        <span
          class="fui-Button__icon rywnvv2"
        >
          <span
            aria-hidden="true"
            class="root_dd790ee3 emotion-1"
          >
            <svg
              class="svg_dd790ee3"
              focusable="false"
              viewBox="0 0 2048 2048"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1024 0q141 0 272 36t244 104 207 160 161 207 103 245 37 272q0 141-36 272t-104 244-160 207-207 161-245 103-272 37q-141 0-272-36t-244-104-207-160-161-207-103-245-37-272q0-141 36-272t104-244 160-207 207-161T752 37t272-37zm0 1920q124 0 238-32t214-90 181-140 140-181 91-214 32-239q0-124-32-238t-90-214-140-181-181-140-214-91-239-32q-124 0-238 32t-214 90-181 140-140 181-91 214-32 239q0 124 32 238t90 214 140 181 181 140 214 91 239 32zm443-1249l-352 353 352 353-90 90-353-352-353 352-90-90 352-353-352-353 90-90 353 352 353-352 90 90z"
              />
            </svg>
          </span>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`create shapshot of when type is success 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 1001;
  margin: 0.5rem;
}

.emotion-1 {
  width: 1rem;
  height: 1rem;
}

<div
    class="emotion-0"
  >
    <div
      class="fui-Alert ___co0mdz0_q48w6e0 f22iagw f122n59 f5pgtk9 f1g0x7ka f11qrl6u f1qch9an fjlbh76 fff7au0 f1bjk9e1 fwsfkhu f8wkphi f5ogflp f1hqa2wf f1f09k3d finvdd3 fzkkow9 fcdblym fg706s2 fjik90z fghlq4f f1gn591s fb073pr fjscplz fz58gqq fkhj508 fl43uef f19n0e5 fxugw4r"
      role="status"
    >
      <span
        class="fui-Alert__icon ___srf8880_13m7jpf fd461yt f4ybsrx f1g0x7ka f19lj068 f1qch9an f1cnd47f f1m7fhi8"
      >
        <svg
          aria-hidden="true"
          class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
          fill="currentColor"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 2a8 8 0 1 1 0 16 8 8 0 0 1 0-16Zm3.36 5.65a.5.5 0 0 0-.64-.06l-.07.06L9 11.3 7.35 9.65l-.07-.06a.5.5 0 0 0-.7.7l.07.07 2 2 .07.06c.17.11.4.11.56 0l.07-.06 4-4 .07-.08a.5.5 0 0 0-.06-.63Z"
            fill="currentColor"
          />
        </svg>
      </span>
      success
      <button
        class="fui-Button rlr4yyk fui-Alert__action ___jw5j7g0_17p02te f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu f16muhyy fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 fwa4z56 fe5wyld ffrv2ww f1rnfj6r f9dg1pr f4xjyn1 f19ot1t3 fjaevkr f1yrt24w f1lym4af f11cxmw1 f9ddjv3 f1sbtcvk f81rol6 fdghr9 frdkuqy fy77jfu f44c6la fcgxt0o"
        title="Dismiss"
        type="button"
      >
        <span
          class="fui-Button__icon rywnvv2"
        >
          <span
            aria-hidden="true"
            class="root_dd790ee3 emotion-1"
          >
            <svg
              class="svg_dd790ee3"
              focusable="false"
              viewBox="0 0 2048 2048"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1024 0q141 0 272 36t244 104 207 160 161 207 103 245 37 272q0 141-36 272t-104 244-160 207-207 161-245 103-272 37q-141 0-272-36t-244-104-207-160-161-207-103-245-37-272q0-141 36-272t104-244 160-207 207-161T752 37t272-37zm0 1920q124 0 238-32t214-90 181-140 140-181 91-214 32-239q0-124-32-238t-90-214-140-181-181-140-214-91-239-32q-124 0-238 32t-214 90-181 140-140 181-91 214-32 239q0 124 32 238t90 214 140 181 181 140 214 91 239 32zm443-1249l-352 353 352 353-90 90-353-352-353 352-90-90 352-353-352-353 90-90 353 352 353-352 90 90z"
              />
            </svg>
          </span>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`create shapshot of when type is warning 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 1001;
  margin: 0.5rem;
}

.emotion-1 {
  width: 1rem;
  height: 1rem;
}

<div
    class="emotion-0"
  >
    <div
      class="fui-Alert ___co0mdz0_q48w6e0 f22iagw f122n59 f5pgtk9 f1g0x7ka f11qrl6u f1qch9an fjlbh76 fff7au0 f1bjk9e1 fwsfkhu f8wkphi f5ogflp f1hqa2wf f1f09k3d finvdd3 fzkkow9 fcdblym fg706s2 fjik90z fghlq4f f1gn591s fb073pr fjscplz fz58gqq fkhj508 fl43uef f19n0e5 fxugw4r"
      role="alert"
    >
      <span
        class="fui-Alert__icon ___znwrb80_14kh1uj fd461yt f4ybsrx f1g0x7ka f19lj068 f1qch9an f1cnd47f fpti2h4"
      >
        <svg
          aria-hidden="true"
          class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
          fill="currentColor"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.68 2.79a1.5 1.5 0 0 1 2.64 0l6.5 12A1.5 1.5 0 0 1 16.5 17h-13a1.5 1.5 0 0 1-1.32-2.21l6.5-12ZM10.5 7.5a.5.5 0 0 0-1 0v4a.5.5 0 0 0 1 0v-4Zm.25 6.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z"
            fill="currentColor"
          />
        </svg>
      </span>
      warning
      <button
        class="fui-Button rlr4yyk fui-Alert__action ___jw5j7g0_17p02te f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu f16muhyy fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 fwa4z56 fe5wyld ffrv2ww f1rnfj6r f9dg1pr f4xjyn1 f19ot1t3 fjaevkr f1yrt24w f1lym4af f11cxmw1 f9ddjv3 f1sbtcvk f81rol6 fdghr9 frdkuqy fy77jfu f44c6la fcgxt0o"
        title="Dismiss"
        type="button"
      >
        <span
          class="fui-Button__icon rywnvv2"
        >
          <span
            aria-hidden="true"
            class="root_dd790ee3 emotion-1"
          >
            <svg
              class="svg_dd790ee3"
              focusable="false"
              viewBox="0 0 2048 2048"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1024 0q141 0 272 36t244 104 207 160 161 207 103 245 37 272q0 141-36 272t-104 244-160 207-207 161-245 103-272 37q-141 0-272-36t-244-104-207-160-161-207-103-245-37-272q0-141 36-272t104-244 160-207 207-161T752 37t272-37zm0 1920q124 0 238-32t214-90 181-140 140-181 91-214 32-239q0-124-32-238t-90-214-140-181-181-140-214-91-239-32q-124 0-238 32t-214 90-181 140-140 181-91 214-32 239q0 124 32 238t90 214 140 181 181 140 214 91 239 32zm443-1249l-352 353 352 353-90 90-353-352-353 352-90-90 352-353-352-353 90-90 353 352 353-352 90 90z"
              />
            </svg>
          </span>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
